src/main/fc/rc.c

```diff
@@ -148,6 +148,24 @@ float applyRaceFlightRates(const int axis, float rcCommandf, const float rcComma
     return angleRate;
 }
 
+float applyKissRates(const int axis, float rcCommandf, const float rcCommandfAbs)
+{
+    float kissRpyUseRates;
+    const float rcCurvef = currentControlRateProfile->rcExpo[axis] / 100.0f;
+
+    if (currentControlRateProfile->rates[axis] < 100) {
+        kissRpyUseRates = 1 - rcCommandfAbs * currentControlRateProfile->rates[axis] / 100.0f;
+    } else {
+        //prevent division by zero
+        kissRpyUseRates = 1 - rcCommandfAbs * 0.99f;
+    }
```

Comment:
Seems like it would be better to precalculate the `(1.0f / kissRpyUseRates)` here and for the else clause you could just avoid the division by 0. Rather than using som `0.99` logic.

Reply:
ok